From ac9242a3f074880000d8d81632b4c12ed50cffb0 Mon Sep 17 00:00:00 2001
From: saravananm <saravananm@ami.com>
Date: Mon, 9 Dec 2024 23:45:17 +0530
Subject: [PATCH] Added Fix For SDR Preserve Configuration.

Description:
SDR will not be preserved when PreserveConfiguration SDR is true

Tested:
Build taken successfully and Verified.

Working as Expected.

Signed-off-by: saravananm <saravananm@ami.com>
---
 src/entity_manager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/entity_manager.cpp b/src/entity_manager.cpp
index 3325016..a523291 100644
--- a/src/entity_manager.cpp
+++ b/src/entity_manager.cpp
@@ -1384,7 +1384,7 @@ int main()
     });
     tryIfaceInitialize(entityIface);
 
-    if (fwVersionIsSame())
+    if (fwVersionIsSame() || std::filesystem::exists(currentConfiguration))
     {
         if (std::filesystem::is_regular_file(currentConfiguration))
         {
-- 
2.34.1

